Validation split added: custom data files @sgugger, @patil-suraj

Souvic opened the pull request:
# What does this PR do?
Validation split added in case of no validation file and loading custom data for TensorFlow examples run_mlm.py file
Fixes # (issue)

Issue #12406 fixed.
Docs on language modeling TensorFlow updated.

Souvic committed:
Validation split added: custom data files

Validation split added in case of no validation file and loading custom data

Souvic committed:
Updated documentation with custom file usage

Updated documentation with custom file usage

Souvic commented:
@sgugger 
@patil-suraj

sgugger commented on examples/tensorflow/language-modeling/README.md:
````diff
@@ -48,6 +48,15 @@ python run_mlm.py \
 --dataset_config_name wikitext-103-raw-v1
 ```
+```
+#In case using a custom dataset
+#validation file can also be added
+#Otherwise some split(customizable) of training data is used as validation
````
This can be put in text before the ```

sgugger reviewed (approved):
Thanks a lot for adding this! I think you forgot to deal with the `run_clm` file however. Once this is done, you should run `make style` on your branch to apply the formatter to your code.

Souvic committed:
Update README.md

Souvic committed:
Update README.md

Souvic committed:
Update README.md

Souvic committed:
Made some suggested stylistic changes

Souvic committed:
Used logger instead of print.

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Souvic committed:
Made similar changes to add validation split

In case of a missing validation file, a validation split will be used now.

Souvic committed:
max_train_samples to be used for training only

max_train_samples got misplaced, now corrected so that it is applied on training data only, not whole data.

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic committed:
styled

Souvic commented:
@sgugger
@patil-suraj
Made the necessary changes.. please see and comment.

sgugger commented on examples/tensorflow/language-modeling/README.md:
````diff
@@ -49,6 +49,15 @@ python run_mlm.py \
 --dataset_config_name wikitext-103-raw-v1
 ```
 
+In case using a custom dataset, validation file can be seperately passed as an input argument.\
+Otherwise some split(customizable) of training data is used as validation.
````
```suggestion
When using a custom dataset, the validation file can be separately passed as an input argument. Otherwise some split (customizable) of training data is used as validation.
```

sgugger commented on examples/tensorflow/language-modeling/README.md:
````diff
@@ -61,3 +70,13 @@ python run_clm.py \
 --dataset_config_name wikitext-103-raw-v1
 ```
+
+In case using a custom dataset, validation file can be seperately passed as an input argument.\
+Otherwise some split(customizable) of training data is used as validation.
````
```suggestion
When using a custom dataset, the validation file can be separately passed as an input argument. Otherwise some split (customizable) of training data is used as validation.
```

Souvic replied:
circleci is failing on code quality check...
Order of imports may not be correct. But I guess you kept it that way to improve readability.. 
Not sure. I tried changing one of the files. It is still failing.

sgugger replied:
It's nothing to do with those comments, you have to run `make style` on your branch to fix the import order in both scripts. Make sure you have installed the proper versions of our quality tools with `pip install -e .[quality]` in the transformers repo.

Souvic replied:
Yes yes.. not those comments. I was not sure why it was happening.
I installed black myself I guess. Correct pip installing as you suggested fixed the issue. Thanks!

sgugger reviewed (approved):
Thanks, just a last nit and we should be good to merge.

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic committed:
changed ordering

Souvic committed:
Improved language of documentation

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Souvic committed:
Improved language of documentation

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic committed:
Fixed styling issue

Souvic committed:
Update run_mlm.py

Souvic committed:
Merge branch 'huggingface:master' into patch-3

Souvic commented:
Hopefully done now!

sgugger commented:
All good now, thanks!
